src/MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -56,34 +60,51 @@ ErrorMessageOr<MetricsUploader> MetricsUploader::CreateMetricsUploader(std::stri
     return ErrorMessage("MetricsUploader is already created");
   }
 
+  // Generate UUID as a session identifier.
+  UUID uuid;
+  RPC_STATUS create_status = UuidCreate(&uuid);
+  if (create_status != RPC_S_OK) {
+    return ErrorMessage("Unable to create UUID for metrics uploader");
+  }
+  RPC_CSTR uuid_c_str = nullptr;
+  RPC_STATUS convert_status = UuidToStringA(&uuid, &uuid_c_str);
+  if (convert_status != RPC_S_OK) {
+    return ErrorMessage("Unable to convert UUID to string for metrics uploader");
+  }
+
+  std::string session_uuid{reinterpret_cast<char*>(uuid_c_str)};
```

Comment:
uuid_c_str is a null terminated `unsigned char*`. I tried static cast, but it did not work out of the box. Any other ideas?

Reply:
Ah I see. RPC_CSTR is probably `unsigned char*` under the hood. So that's unfortunate. It's probably best to leave the reinterpret_cast, but to protect it against misuse. The problem with reinterpret_cast is that it would also work if the type of RPC_CSTR changed. So I would add a static_assert which ensures that RPC_CSTR is of type `usigned char*`.

```cpp
#include <type_traits>
static_assert(std::is_same_v<RPC_CSTR, unsigned char*>, "The type of RPC_CSTR needs to be castable to const char*");
```

Reply:
Obviously the type of RPC_CSTR will never change. This is more to show others that you thought about potential problems with reinterpret_cast. A comment (or both) could also be enough.

Reply:
This was a great idea! Thank you, I added it